src/explore-education-statistics-admin/src/hooks/useReleaseServiceStatus.ts

```diff
@@ -0,0 +1,84 @@
+import { StatusBlockProps } from '@admin/components/StatusBlock';
+import { getStatusDetail } from '@admin/components/ReleaseServiceStatus';
+import releaseService, {
+  ReleaseStageStatuses,
+} from '@admin/services/releaseService';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { forceCheck } from 'react-lazyload';
+
+export interface StatusDetail {
+  color: StatusBlockProps['color'];
+  text: string;
+}
+
+interface Props {
+  refreshPeriod?: number;
+  releaseId: string;
+}
+
+/**
+ * Hook to get and refresh the status for the given release.
+ */
+export default function useReleaseServiceStatus({
```

Comment:
Similar to an earlier component, a more appropriate name would be `useReleasePublishingStatus`